include/boost/geometry/algorithms/detail/disjoint/segment_box.hpp

```diff
@@ -186,7 +186,17 @@ struct disjoint_segment_box_sphere_or_spheroid
         bool b2 = formula::azimuth_side_value(alp1, a_b2) > 0;
         bool b3 = formula::azimuth_side_value(alp1, a_b3) > 0;
 
-        if (!(b0 && b1 && b2 && b3) && (b0 || b1 || b2 || b3))
+        bool all_ones = b0 && b1 && b2 && b3;
+        bool all_zeros = !(b0 || b1 || b2 || b3);
+        bool vertex_north = lat1 + lat2 > 0;
+
+        if ((all_ones && vertex_north) || (all_zeros && !vertex_north))
+        {
+            return disjoint_info::disjoint_no_vertex;
+        }
+
+        if ((!all_ones && (b0 || b1 || b2 || b3) && vertex_north) ||
+            (!all_zeros && !(b0 && b1 && b2 && b3) && !vertex_north))
```

Comment:
For now leaving the fact that `all_zeros` can be `true` if side values are `-1` this expression could be simplified to:
```
(!all_ones && !all_zeros && vertex_north) || (!all_zeros && all_ones && !vertex_north)
```

Reply:
nice catch! actually 
```
(!all_ones && !all_zeros && vertex_north) || (!all_zeros && !all_ones && !vertex_north)
```
(there was a typo above) 
Which can be further simplified to `(!all_ones && !all_zeros)`

I also added the special case of some side value is equal to `0`.